File: app/src/main/java/com/kfang/fencemelibrary/main/SharedPreferencesRepositoryImpl.java
package com.kfang.fencemelibrary.main;

import android.content.SharedPreferences;
import android.os.Vibrator;

import static com.kfang.fencemelibrary.misc.Constants.BOUT_LENGTH_MINUTES;
import static com.kfang.fencemelibrary.misc.Constants.BOUT_LENGTH_POINTS;
import static com.kfang.fencemelibrary.misc.Constants.DEFAULT_MINUTES;
import static com.kfang.fencemelibrary.misc.Constants.DEFAULT_POINTS;
import static com.kfang.fencemelibrary.misc.Constants.KEEP_DEVICE_AWAKE;
import static com.kfang.fencemelibrary.misc.Constants.PAUSE_ON_SCORE_CHANGE;
import static com.kfang.fencemelibrary.misc.Constants.POPUP_ON_SCORE;
import static com.kfang.fencemelibrary.misc.Constants.RESTORE_ON_EXIT;
import static com.kfang.fencemelibrary.misc.Constants.TOGGLE_DOUBLE_TOUCH;
import static com.kfang.fencemelibrary.misc.Constants.VIBRATE_AT_END;
import static com.kfang.fencemelibrary.misc.Constants.VIBRATE_TIMER;

/**
 * SharedPreferences repository implementation
 */

public class SharedPreferencesRepositoryImpl implements SharedPreferencesRepository {

    private SharedPreferences prefs;
    private Vibrator v;

    public SharedPreferencesRepositoryImpl(SharedPreferences sp, Vibrator v) {
        prefs = sp;
        this.v = v;
    }

    @Override
    public boolean stayAwakeDuringTimer() {
        return prefs.getBoolean(KEEP_DEVICE_AWAKE, true);
    }

    @Override
    public int pointsToWin() {
        return prefs.getInt(BOUT_LENGTH_POINTS, DEFAULT_POINTS);
    }

    @Override
    public boolean popupOnScoreChange() {
        return prefs.getBoolean(POPUP_ON_SCORE, false);
    }

    @Override
    public boolean restoreOnAppReset() {
        return prefs.getBoolean(RESTORE_ON_EXIT, true);
    }

    @Override
    public boolean vibrateOnTimerFinish() {
        return prefs.getBoolean(VIBRATE_AT_END, true);
    }

    @Override
    public boolean enableDoubleTouch() {
        return prefs.getBoolean(TOGGLE_DOUBLE_TOUCH, true);
    }

    @Override
    public boolean vibrateOnTimerToggle() {
        return prefs.getBoolean(VIBRATE_TIMER, true) && v.hasVibrator();
    }

    @Override
    public boolean pauseOnScoreChange() {
        return prefs.getBoolean(PAUSE_ON_SCORE_CHANGE, true);
    }

    @Override
    public int getBoutLengthMinutes() {
        return prefs.getInt(BOUT_LENGTH_MINUTES, DEFAULT_MINUTES);
    }
}

File: app/src/main/java/com/kfang/fencemelibrary/presentation/SharedPreferencesRepositoryImpl.kt
package com.kfang.fencemelibrary.presentation

import android.content.SharedPreferences
import android.os.Vibrator
import com.kfang.fencemelibrary.misc.Constants

/**
 * SharedPreferences repository implementation
 */

internal class SharedPreferencesRepositoryImpl(private val prefs: SharedPreferences, private val v: Vibrator) : SharedPreferencesRepository {

    override fun stayAwakeDuringTimer(): Boolean {
        return prefs.getBoolean(Constants.KEEP_DEVICE_AWAKE, true)
    }

    override fun pointsToWin(): Int {
        return prefs.getInt(Constants.BOUT_LENGTH_POINTS, Constants.DEFAULT_POINTS)
    }

    override fun popupOnScoreChange(): Boolean {
        return prefs.getBoolean(Constants.POPUP_ON_SCORE, false)
    }

    override fun restoreOnAppReset(): Boolean {
        return prefs.getBoolean(Constants.RESTORE_ON_EXIT, true)
    }

    override fun vibrateOnTimerFinish(): Boolean {
        return prefs.getBoolean(Constants.VIBRATE_AT_END, true)
    }

    override fun volumeButtonTimerToggle(): Boolean {
        return prefs.getBoolean(Constants.VOLUME_BUTTON_TIMER_TOGGLE, true)
    }

    override fun enableDoubleTouch(): Boolean {
        return prefs.getBoolean(Constants.TOGGLE_DOUBLE_TOUCH, true)
    }

    override fun vibrateOnTimerToggle(): Boolean {
        return prefs.getBoolean(Constants.VIBRATE_TIMER, true) && v.hasVibrator()
    }

    override fun pauseOnScoreChange(): Boolean {
        return prefs.getBoolean(Constants.PAUSE_ON_SCORE_CHANGE, true)
    }

    override fun getBoutLengthMinutes(): Int {
        return prefs.getInt(Constants.BOUT_LENGTH_MINUTES, Constants.DEFAULT_MINUTES)
    }
}
